.github/workflows/workflow.yml

```diff
@@ -72,8 +69,37 @@ jobs:
         use-verbose-mode: 'yes'
         config-file: '.md-link-check.json' 
 
-  integration:
+  unit_tests:
+    name: Unit Tests
+    needs:
+    - code_quality_check
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+
+    timeout-minutes: 30
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: pip install tox==3.25.1
+    - name: Unit Tests
+      run: |
+        tox -e test-unit
+      shell: bash
+    - name: Coverage Report
+      run: tox -e coverage-report
+
+  integration_tests:
     name: Integration tests
+    needs:
+    - unit_tests
     runs-on: ${{ matrix.os }}
```

Comment:
just change to ubuntu-latest

Reply:
it's correct, it runs on three platforms